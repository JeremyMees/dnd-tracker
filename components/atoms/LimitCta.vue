<script setup lang="ts">
const isShown = ref<boolean>(false)
const cookie = useCookie<number>('limit_cta')
const removeTimer = ref()
const day = 24 * 60 * 60 * 1000

defineExpose({ show })

onMounted(() => {
  setTimeout(() => checkShow(), 50)
  setAutoCloseTimer()
})

function setAutoCloseTimer(): void {
  removeTimer.value = setTimeout(() => {
    isShown.value = false
  }, 10000)
}

function show(): void {
  if (removeTimer.value) clearTimeout(removeTimer.value)

  isShown.value = true
  cookie.value = Date.now()

  setAutoCloseTimer()
}

function checkShow(): void {
  if (!cookie.value || isNaN(cookie.value)) show()
  else {
    const now = Date.now()
    if (now - cookie.value > day) show()
  }
}
</script>

<template>
  <AnimationExpand>
    <div
      v-if="isShown"
      class="fixed bottom-2 left-1/2 -translate-x-1/2 z-10 w-full max-w-prose"
    >
      <Card
        color="info"
        class="flex flex-col md:flex-row md:items-center gap-4 backdrop-blur-xl"
      >
        <button
          class="absolute top-0 right-0 group"
          :aria-label="$t('actions.close')"
          @click="isShown=false"
        >
          <Icon
            name="tabler:x"
            class="text-destructive size-4 rounded-full ring-destructive group-focus-within:ring"
            aria-hidden="true"
          />
        </button>
        <p class="body-small">
          {{ $t('components.limitCta.text') }}
        </p>
        <div class="flex justify-end">
          <NuxtLinkLocale
            to="/pricing"
            class="btn-primary whitespace-nowrap"
          >
            {{ $t('components.limitCta.cta') }}
          </NuxtLinkLocale>
        </div>
      </Card>
    </div>
  </AnimationExpand>
</template>
