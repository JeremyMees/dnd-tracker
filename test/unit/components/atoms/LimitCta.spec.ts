import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'
import LimitCta from '~/components/atoms/LimitCta.vue'

describe('LimitCta', async () => {
  it('Should match snapshot', async () => {
    const component = await mountSuspended(LimitCta)
    expect(component.html()).toMatchSnapshot()
  })
})
