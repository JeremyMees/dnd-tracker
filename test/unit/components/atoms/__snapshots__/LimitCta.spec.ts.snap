// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LimitCta > Should match snapshot 1`] = `""`;
